Add title section to customization page (#2838)
altair-viz/altair: doc/user_guide/customization.rst

just before the chart is rendered.

Adjusting the Title
-------------------
By default an Altair chart does not have a title, as seen in this example.

.. altair-plot::

   import altair as alt
   from vega_datasets import data
   
   iowa = data.iowa_electricity.url
   
   alt.Chart(iowa).mark_area().encode(
       x="year:T",
       y=alt.Y("net_generation:Q", stack="normalize"),
       color="source:N"
   )

You can add a simple title by passing the `title` keyword argument with the data.

.. altair-plot::

   alt.Chart(iowa, title="Iowa's green energy boom").mark_area().encode(
       x="year:T",
       y=alt.Y("net_generation:Q", stack="normalize"),
       color="source:N"
   )

It is also possible to add a subtitle, and `configure <https://altair-viz.github.io/user_guide/configuration.html#config-title>`_ a number of other attributes, by passing in an `alt.Title` object.

.. altair-plot::

   alt.Chart(
      iowa,

>       title=alt.Title("Iowa's green energy boom", subtitle="A growing share of the state's energy has come from renewable sources")

Reviewer: ```suggestion
      title=alt.Title(
          "Iowa's green energy boom",
          subtitle="A growing share of the state's energy has come from renewable sources"
      )
```

just before the chart is rendered.

Adjusting the Title
-------------------
By default an Altair chart does not have a title, as seen in this example.

.. altair-plot::

   import altair as alt
   from vega_datasets import data
   
   iowa = data.iowa_electricity.url
   
   alt.Chart(iowa).mark_area().encode(
       x="year:T",
       y=alt.Y("net_generation:Q", stack="normalize"),
       color="source:N"
   )

You can add a simple title by passing the `title` keyword argument with the data.

.. altair-plot::

   alt.Chart(iowa, title="Iowa's green energy boom").mark_area().encode(
       x="year:T",
       y=alt.Y("net_generation:Q", stack="normalize"),
       color="source:N"
   )

It is also possible to add a subtitle by passing in an `alt.TitleParams` object.

.. altair-plot::

   alt.Chart(
      iowa,

>       title=alt.TitleParams(

Reviewer: ```suggestion
      title=alt.Title(
```

          "Iowa's green energy boom",
          subtitle="A growing share of the state's energy has come from renewable sources"
      )
   ).mark_area().encode(
       x="year:T",
       y=alt.Y("net_generation:Q", stack="normalize"),
       color="source:N"
   )

> The subtitle can run to two lines.

Reviewer: ```suggestion
The subtitle can run to two lines by passing a list where each list item is a line (if you don't want to create this list manually as in the example below, you can use the ``wrap`` function from the `textwrap library https://docs.python.org/3/library/textwrap.html`_ to split a string into a list of substrings of a certain length).
```

.. altair-plot::

   alt.Chart(
      iowa,

>       title=alt.TitleParams(

Reviewer: ```suggestion
      title=alt.Title(
```

          "Iowa's green energy boom",
          subtitle=["A growing share of the state's energy", "has come from renewable sources"]
      )
   ).mark_area().encode(
       x="year:T",
       y=alt.Y("net_generation:Q", stack="normalize"),
       color="source:N"
   )

> The parameters object can also `configure <https://altair-viz.github.io/user_guide/configuration.html#config-title>`_ a number of other attributes.

Reviewer: ```suggestion
The ``Title`` object can also configure a number of other attributes, e.g., the position of the title and subtitle (see see :ref:`user-guide-customization` for details).
```

.. altair-plot::

   alt.Chart(
      iowa,

>       title=alt.TitleParams(

Reviewer: ```suggestion
      title=alt.Title(
```
